src/v/cluster/controller_backend.cc

```diff
@@ -545,8 +555,102 @@ bool is_finishing_operation(
       [](const auto&) { return false; });
 }
 
+/**
+ * Retrieve topic property based on the following logic
+ *
+ *
+ * +---------------------------------+---------------+----------+-------------+
+ * |Cluster(optional)\Topic(tristate)|     Empty     | Disabled |    Value    |
+ * +---------------------------------+---------------+----------+-------------+
+ * |Empty                            | OFF           | OFF      | Topic Value |
+ * |Value                            | Cluster Value | OFF      | Topic Value |
+ * +---------------------------------+---------------+----------+-------------+
+ *
+ */
+template<typename T>
+std::optional<T> get_topic_property(
+  std::optional<T> cluster_level_property, tristate<T> topic_property) {
+    // disabled
+    if (topic_property.is_disabled()) {
+        return std::nullopt;
+    }
+    // has value
+    if (topic_property.has_optional_value()) {
+        return *topic_property;
+    }
```

Comment:
If the tristate is `Disabled` shouldn't we honour that?

Reply:
i think it is handled few lines before this